package.json

```diff
@@ -130,10 +131,53 @@
     "shx": "^0.3.2",
     "sloc": "^0.3.1",
     "sort-package-json": "^2.8.0",
-    "typescript": "~5.2.0"
+    "typescript": "~5.2.0",
+    "wireit": "^0.14.9"
   },
   "packageManager": "yarn@3.8.1",
   "engines": {
     "node": "18 || 20"
+  },
+  "wireit": {
+    "prettier:check": {
+      "command": "node ./scripts/log-output.js --log ./.tasks/prettier-check.log -- prettier --check .",
+      "files": [
+        "**/*.{.dockerbuild,.dockerfile,css,html,js,json,jsx,md,mdx,mustache,ps1,scss,sh,ts,tsx,xml,yaml,yml}"
+      ],
+      "output": [
+        "./.tasks/prettier-check.log"
+      ]
+    },
+    "prettier:fix": {
+      "command": "node ./scripts/log-output.js --log ./.tasks/prettier-fix.log -- prettier --write .",
+      "files": [
+        "**/*.{.dockerbuild,.dockerfile,css,html,js,json,jsx,md,mdx,mustache,ps1,scss,sh,ts,tsx,xml,yaml,yml}"
+      ],
+      "output": [
+        "./.tasks/prettier-fix.log"
+      ]
+    },
+    "tsc:simple": {
+      "clean": "if-file-deleted",
+      "command": "tsc",
+      "files": [
+        "**/*.{ts,tsx}",
+        "**/tsconfig.json"
+      ],
+      "output": [
+        "./dist-types"
+      ]
+    },
+    "tsc:full": {
+      "clean": "if-file-deleted",
+      "command": "backstage-cli repo clean && tsc --skipLibCheck false --incremental false",
+      "files": [
+        "**/*.{ts,tsx}",
+        "**/tsconfig.json"
+      ],
```

Comment:
`tsc` needs to be run for `.js` and its flavors too, since we have some type checking there as well, and `package.json` changes can affect types too. More importantly though, does this run when there are changes in `node_modules`? That would need to be the case. For CI we could check `yarn.lock` for changes, although I assume it's a bit trickier locally.